src/main/java/seedu/address/logic/parser/ParserUtil.java

```diff
@@ -149,8 +149,16 @@ public static Attendance parseAttendance(String input) throws ParseException {
         try {
             return Attendance.fromDateString(input);
         } catch (IllegalArgumentException e) {
-            throw new ParseException(Attendance.MESSAGE_CONSTRAINTS);
+            throw parseException(input, Attendance.MESSAGE_CONSTRAINTS);
         }
     }
 
+    private static String inputIsWrong(String input, String message) {
+        return "Your input: \"" + input + "\" is incorrectly formatted.\n"
+                + "See below for further details:\n\n" + message;
+    }
+
+    private static ParseException parseException(String input, String message) {
+        return new ParseException(inputIsWrong(input, message));
```

Comment:
Should this be `throw new ParseException?`Not that its a problem, usually thought we throw exceptions.

Reply:
nah I want to keep the `throw` within each function to be explicit the functions throw exceptions.

Otherwise it won't be apparent where the exception occurs.

```
    public static Attendance parseAttendance(String input) throws ParseException {
        requireNonNull(input);
        try {
            return Attendance.fromDateString(input);
        } catch (IllegalArgumentException e) {
            parseException(input, Attendance.MESSAGE_CONSTRAINTS); // This is not explicit
        }
    }
```